Helper/DocumentImporter.php

```diff
@@ -0,0 +1,199 @@
+    private $environment;
+    /** @var int */
+    private $bulkSize;
+    /** @var string  */
+    private $instanceId;
+    /** @var bool */
+    private $finalize;
+    /** @var bool */
+    private $force;
+
+    public function __construct(DataService $dataService, EntityManager $entityManager, FormFactoryInterface $formFactory, Bulker $bulker, string $instanceId, string $contentTypeName, string $lockUser, bool $rawImport, bool $signData, bool $indexInDefaultEnv, int $bulkSize, bool $finalize, bool $force)
+    {
+        $this->contentTypeName = $contentTypeName;
+        $this->indexInDefaultEnv = $indexInDefaultEnv;
+        $this->instanceId = $instanceId;
+        $this->signData = $signData;
+        $this->lockUser = $lockUser;
+        $this->rawImport = $rawImport;
+        $this->bulkSize = $bulkSize;
+        $this->finalize = $finalize;
+        $this->dataService = $dataService;
+        $this->bulker = $bulker;
+        $this->entityManager = $entityManager;
+        $this->formFactory = $formFactory;
+        $this->force = $force;
+
+        //https://stackoverflow.com/questions/9699185/memory-leaks-symfony2-doctrine2-exceed-memory-limit
+        //https://stackoverflow.com/questions/13093689/is-there-a-way-to-free-the-memory-allotted-to-a-symfony-2-form-object
+        $this->entityManager->getConnection()->getConfiguration()->setSQLLogger(null);
+
+        $this->bulker->setEnableSha1(false);
+
+
+        $repository = $entityManager->getRepository('EMSCoreBundle:Revision');
+        if (! $repository instanceof RevisionRepository) {
+            throw new \Exception('Can not get the RevisionRepository');
+        }
+        $this->revisionRepository = $repository;
+
+        $repository = $entityManager->getRepository('EMSCoreBundle:ContentType');
+        if (! $repository instanceof ContentTypeRepository) {
+            throw new \Exception('Can not get the ContentTypeRepository');
+        }
+        $this->contentTypeRepository = $repository;
+
+        $this->bulker->setSize($this->bulkSize);
+
+        $this->initEntities();
+    }
+
+    public function clearAndSend($lastCall = false)
+    {
+        $this->entityManager->flush();
+        $this->revisionRepository->clear();
+        $this->contentTypeRepository->clear();
+        $this->entityManager->clear();
+        unset($this->environment);
+        unset($this->contentType);
+
+        $this->bulker->send(true);
+
+        if (!$lastCall) {
+            $this->initEntities();
+        }
+    }
+
+    private function initEntities()
+    {
+        $contentType = $this->contentTypeRepository->findOneBy(array("name" => $this->contentTypeName, 'deleted' => false));
+        if (! $contentType instanceof ContentType) {
+            throw new \Exception(sprintf('Content type %s not found', $this->contentTypeName));
+        }
+        $this->contentType = $contentType;
+        $this->environment = $this->contentType->getEnvironment();
```

Comment:
should be part of the contextual object

Reply:
1e8555d195f4caf512272f5f4efed1125cc64771